todo_app/data/trello_items.py

```diff
@@ -0,0 +1,176 @@
+
+    response = requests.request("GET", url, headers=headers, params=query, timeout=10)
+
+    lists = json.loads(response.text)
+    items = []
+
+    # For now let's just add all items sorted according to status and then by ID
+    # If needed, we can sort by ID later
+    for cList in lists:
+        if cList['name'] not in ['To Do', 'Doing']:
+            continue
+        for card in cList['cards']:
+            items.append(TDItem.fromTrelloCards(card, cList))
+
+    return items
+
+def getItem(itemID):
+    """
+    Fetches the saved item with the specified ID.
+
+    Args:
+        id: The ID of the item.
+
+    Returns:
+        item: The saved item, or None if no items match the specified ID.
+    """
+    items = getItems()
+    return next((item for item in items if item.id == int(itemID)), None)
+
+def addItem(title):
+    """
+    Adds a new item with the specified title to the session.
+
+    Args:
+        title: The title of the item.
+
+    Returns:
+        item: The saved item.
+    """
+    # First need to get the ID of the lists
+    url = f'https://api.trello.com/1/boards/{os.environ.get("TRELLO_BOARD")}/lists'
+    headers = {"Accept": "application/json"}
+    query = {'key': os.environ.get("TRELLO_KEY"), 'token': os.environ.get("TRELLO_TOKEN")}
+
+    response = requests.request("GET", url, headers=headers, params=query, timeout=10)
+    lists = {entry['name']: entry['id'] for entry in json.loads(response.text)}
+
+    # Now add the new item to the To Do list
+    # Can also check in the future if the item is Doing, and choose the appropriate list
+    url = "https://api.trello.com/1/cards"
+
+    headers = {"Accept": "application/json"}
+    query = {'idList': lists['To Do'], 'name': title,
+             'key': os.environ.get("TRELLO_KEY"), 'token': os.environ.get("TRELLO_TOKEN")}
+
+    response = requests.request("POST", url, headers=headers, params=query, timeout=10)
+    newCard = json.loads(response.text)
+
+    return {'id': newCard['id'], 'status': 'To Do', 'title': newCard['name']}
+
+def saveItem(item):
+    """
+    Updates an existing item in the session. If no existing item matches the ID of the specified item, nothing is saved.
+
+    Args:
+        item: The item to save.
+    """
+    url = f'https://api.trello.com/1/boards/{os.environ.get("TRELLO_BOARD")}/lists'
+    headers = {"Accept": "application/json"}
+    query = {'cards': 'open', 'key': os.environ.get("TRELLO_KEY"), 'token': os.environ.get("TRELLO_TOKEN")}
+
+    response = requests.request("GET", url, headers=headers, params=query, timeout=10)
+
+    lists = json.loads(response.text)
+    listIDs = {entry['name']: entry['id'] for entry in lists}
+    cards = []
+    for cList in lists:
+        for card in cList['cards']:
+            cards.append(card)
+    cardID = -1
+    for card in cards:
+        if card['idShort'] == item.id:
+            cardID = card['id']
+
+    if cardID == -1:
+        return item
```

Comment:
It doesn't sort through the cards. The thing is that I am using the short ID of the Trello card to display in the app because the long ID would be a bit of an eyesore. So what I am doing is getting all the cards and matching the short ID to a long ID, and using the long ID to call the API, hence why two calls.

But looking at this, I can definitely make an improvement. I can save the long ID along with the short ID in the card object and simply use that. Should be pretty easy to implement.

Reply:
Actually, won't be so easy, cause I am getting the cards on the fly and not actually storing them locally... Will have to leave this for the future, if I ever change the internal data model.